### PR TITLE
[code-generator] feat: add func for generating register code

### PR DESCRIPTION
**Pull request by i0r3k**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
The kube_codegen.sh of code-generator lacks function for generating register code, it'd be more convenient to provide it for users. 

#### Which issue(s) this PR fixes:

N/A.

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by cici37** [comment]
/assign @sttts 
If you have time to review it. Thanks :)

/triage accepted

**Comment by sttts** [comment]
Sgtm. Delegating to @thockin for bash code review.

**Comment by thockin** [comment]
ACK receipt - I will probably not get to this until after KEP freeze, if that's OK.

File: staging/src/k8s.io/code-generator/kube_codegen.sh [hunk 1]
```diff
@@ -665,3 +665,95 @@ function kube::codegen::gen_client() {
             "${input_pkgs[@]}"
     fi
 }
+
+# Generate register code
+#
+# USAGE: kube::codegen::gen_registers [FLAGS] <input-dir>
```
Comment (thockin) [inline staging/src/k8s.io/code-generator/kube_codegen.sh:671]: Can we make this singular?  `gen_register` ?

File: staging/src/k8s.io/code-generator/kube_codegen.sh [hunk 2]
```diff
@@ -665,3 +665,95 @@ function kube::codegen::gen_client() {
...
+
+    # Register
+    #
+    local input_pkgs=()
+    while read -r dir; do
+        pkg="$(cd "${dir}" && GO111MODULE=on go list -find .)"
+        input_pkgs+=("${pkg}")
+    done < <(
+        ( kube::codegen::internal::grep -l --null \
+            -e '+groupName=' \
```
Comment (thockin) [inline staging/src/k8s.io/code-generator/kube_codegen.sh:736]: All of th eother greps have been changed to something like `-e '^\s*//\s*+genclient'` - I assume something similar applies here?

**Review by thockin — COMMENTED** [review]
Just minor picks

File: staging/src/k8s.io/code-generator/kube_codegen.sh (hunk 1 above)
Comment (reaver-flomesh) [reply staging/src/k8s.io/code-generator/kube_codegen.sh:671]: Just follow the naming convention in this script, for example:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/code-generator/kube_codegen.sh#L65
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/code-generator/kube_codegen.sh#L252

File: staging/src/k8s.io/code-generator/kube_codegen.sh (hunk 2 above)
Comment (reaver-flomesh) [reply staging/src/k8s.io/code-generator/kube_codegen.sh:736]: As the Golang source code of register-gen indicates it extracts information from files with `+groupName` comment tag:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/code-generator/cmd/register-gen/generators/targets.go#L84-L90

File: staging/src/k8s.io/code-generator/kube_codegen.sh (hunk 1 above)
Comment (thockin) [reply staging/src/k8s.io/code-generator/kube_codegen.sh:671]: There are several helpers, so plural makes sense.

There is one openapi, so singular makes sense.

Are there multiple registers generated?  I guess there could be?

Comment (reaver-flomesh) [reply staging/src/k8s.io/code-generator/kube_codegen.sh:671]: Got it, will make it singular.

File: staging/src/k8s.io/code-generator/kube_codegen.sh (hunk 2 above)
Comment (thockin) [reply staging/src/k8s.io/code-generator/kube_codegen.sh:736]: I meant that the anchoring could be more specific.  We found some places that were matching on other comments which included `+genclient` but were not actually the declaration for "please generate me a client".  E.g.

We want to match :

```
        // +genclient
```

but not:

```
        // Blah blah blah, and then we include +genclient in the code and blah blah blah
```

**Comment by thockin** [comment]
Looks like you picked up a merge commit.  Try a `git rebase -i` and drop that from your local tree.

**Comment by reaver-flomesh** [comment]
> Looks like you picked up a merge commit. Try a `git rebase -i` and drop that from your local tree.

Ok, no problem.

File: staging/src/k8s.io/code-generator/kube_codegen.sh (hunk 2 above)
Comment (reaver-flomesh) [reply staging/src/k8s.io/code-generator/kube_codegen.sh:736]: Understood

Comment (reaver-flomesh) [reply staging/src/k8s.io/code-generator/kube_codegen.sh]: Fixed as suggested, please take a look if you have any time, thanks. :-)

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve
/retest

**Pull request merged**